quarto/02-QC-AET02.qmd

```diff
@@ -0,0 +1,330 @@
+  variables = c(AEBODSYS),
+  by = c(ARM),
+  overall = TRUE,
+  denominator = adsl,
+  statistic = everything() ~ c("n", "p")) |>
+  dplyr::mutate(tbl_name = 'tot_pat_ae_aebod') |>
+  dplyr::filter(context == "hierarchical") |>
+  dplyr::mutate(
+    stat = purrr::map_dbl(stat, ~ ifelse(length(.x) > 0, as.numeric(.x[[1]]), NA_character_)),
+    group1_level = purrr::map_chr(group1_level, ~ ifelse(length(.x) > 0, as.character(.x[[1]]), NA_character_)))
+
+# Total number of AEs for each AEBODSYS
+tot_ae_aebod <- ard_stack_hierarchical_count(
+  data = adae,
+  variables = c(AEBODSYS),
+  by = c(ARM),
+  denominator = adsl,
+  overall = TRUE) |> 
+  shuffle_ard(trim = FALSE) |>
+  dplyr::mutate(tbl_name = 'tot_ae_aebod') |>
+  dplyr::filter(context == "hierarchical_count") |>
+  dplyr::mutate(
+    stat = purrr::map_dbl(stat, ~ ifelse(length(.x) > 0, as.numeric(.x[[1]]), NA_character_)),
+    ARM = purrr::map_chr(ARM, ~ ifelse(length(.x) > 0, as.character(.x[[1]]), NA_character_)))
+  
+
+# Count and Percent for each AEDECOD within each AEBODSYS
+ae_n_p_aedecod <- ard_stack_hierarchical(
+  data = adae,
+  id = "USUBJID",
+  variables = c(AEBODSYS, AEDECOD),
+  by = c(ARM),
+  denominator = adsl,
+  statistic = everything() ~ c("n", "p"),
+  overall = TRUE) |> 
+  shuffle_ard(trim = FALSE) |> 
+  dplyr::filter(context == "hierarchical", 
+                variable == "AEDECOD") |>
+  dplyr::mutate(tbl_name = 'ae_n_p_aedecod ')|>
+  dplyr::mutate(
+    stat = purrr::map_dbl(stat, ~ ifelse(length(.x) > 0, as.numeric(.x[[1]]), NA_character_)))
+
+# bind the ARDs
+final_aet02_ard <- dplyr::bind_rows(tot_pat_ae, tot_ae, tot_pat_ae_aebod, tot_ae_aebod, ae_n_p_aedecod) |> 
+  as.data.frame() |> 
+  dplyr::select(-c(context, warning, error))
+
+
+final_aet02_ard
+```
+
+
+### Compare Programatically
+
+We'll modify the rtables result to closely match the ARD result:
+1. Rename "group1" and "group1_level"
+2. Remove any columns that aren't in the ARD result (stat_name is being removed as it is a blank column)
+
+```{r}
+
+rtables_result <- rtables_result %>%
+  dplyr::rename(ARM = group2_level)%>%
+  dplyr::select(-c("stat_name", "group2", "stat_string"))
+  
+```
+
+
+Because the subsets of AET02 table are already separated in the ARDs, we can subset the {rtables} result equally to create a clean comparison.
+
+```{r}
+# Total number of patients with at least one adverse event
+tot_pat_ae <- tot_pat_ae |> dplyr::select(c("ARM", "stat"))
+
+tot_pat_ae_rtables <- rtables_result %>%
+  filter(variable_label == "Total number of patients with at least one adverse event" & variable == "USUBJID") |>
+  dplyr::select("ARM", "stat")
+attr(tot_pat_ae_rtables$stat, "label") <- NULL
+
+diffdf(tot_pat_ae, tot_pat_ae_rtables, keys = c("ARM", "stat"))
+
+
+```
+Total number of adverse events
+```{r}
+tot_ae <- tot_ae |>   
+  dplyr::rename(ARM = group1_level) |>
+  dplyr::mutate(ARM = ifelse(is.na(ARM), "Overall ARM", ARM))|>
+  dplyr::select(c("ARM", "stat"))
```

Comment:
Error in `dplyr::rename()` at dplyr/R/mutate.R:146:3:
! Can't rename columns that don't exist.
✖ Column `group1_level` doesn't exist.
Run `rlang::last_trace()` to see where the error occurred.
Error in dplyr::rename(tot_ae, ARM = group1_level) :